### PR TITLE
Fix merge scripts

### PR DESCRIPTION
**Pull request by igordmn**
- `git checkout --no-overlay HEAD --ours` doesn't work with latest Git. Replace it by custom `resetSubFolder` function
- add `x` permission
- add `mergeEmpty.sh` script for merging branch with discarding changes
- add new patterns to `mergeAOSP.sh`

`snapFolder` + `mergeEmpty` usually used together, but I didn't figure yet how to properly combine them into one operation

## Testing
- scripts work as expected

### CODE REVIEW

File: mergeAOSP.sh [hunk 1]
```diff
@@ -15,23 +15,33 @@ COMMIT=$1
...
+# reset subfolder to the HEAD state
+resetSubFolder() {
+    git reset -- "$1" >/dev/null || true       # resets MERGE state
+    git checkout --no-overlay -- "$1" >/dev/null || true    # sets state to the HEAD
+    git clean -fdx -- "$1" >/dev/null || true  # removes new files
+}
+
+resetSubFolder "./buildSrc"
+resetSubFolder "./compose/**/build.gradle"	
+resetSubFolder "./lifecycle/**/build.gradle"
```
Comment (igordmn) [inline mergeAOSP.sh]: New patterns to ignore:
```
"./lifecycle/**/build.gradle"
"./navigation/**/build.gradle"
"./compose/**/nativeMain/**" (conflicts with small renamed files)
```

File: mergeAOSP.sh [hunk 2]
```diff
@@ -15,23 +15,33 @@ COMMIT=$1
 
 git merge $COMMIT --no-commit --no-ff || true
```
Comment (igordmn) [inline mergeAOSP.sh]: I tried to move files into `scripts` folder, but it seems not very convenient to use

Comment (MatkovIvan) [reply mergeAOSP.sh]: The scripts folder was exactly my comment. Root is not a good place for all these scripts. Also, a script shouldn't depend on the current user directory - now it works correctly ONLY if manually/explicitly "`cd`" to root first, but THIS is not convenient and based on assumptions.


The generic pattern is pretty simple:
```sh
ROOT_DIR="$(dirname "$0")/.."
(cd $ROOT_DIR; git merge ...) // Change dir for only one command
```

Or just use `--git-dir` parameter

Comment (igordmn) [reply mergeAOSP.sh]: moved to "scripts"

File: snapSubfolder.sh
Comment (MatkovIvan) [inline snapSubfolder.sh:1]: It seems you forgot to delete a copy from root

Comment (igordmn) [reply snapSubfolder.sh:1]: Fixed

File: scripts/mergeAOSP.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+resetSubFolder "CONTRIBUTING.md"	
+resetSubFolder "README.md"	
+resetSubFolder "gradlew"	
+resetSubFolder "gradlew.bat"	
+resetSubFolder "gradle.properties"	
+resetSubFolder "./gradle/**"	
+resetSubFolder "build.gradle"	
+resetSubFolder "settings.gradle"	
+resetSubFolder "./collection/**/jvmMain/**"	
+resetSubFolder "./collection/**/jvmTest/**"
```
Comment (MatkovIvan) [inline scripts/mergeAOSP.sh:46]:
```
error: pathspec './collection/**/jvmTest/**' did not match any file(s) known to git
```

There is nativeTest instead

Comment (MatkovIvan) [reply scripts/mergeAOSP.sh:46]: Also, it doesn't look like the changes in collection/jvm* should be ignored

**Review by MatkovIvan — APPROVED** [review]

File: scripts/mergeAOSP.sh (hunk 3 above)
Comment (igordmn) [reply scripts/mergeAOSP.sh:46]: It is on purpose. Some folders can be added in the future (actually this folder was in the past, but removed in some commit)

If you know how to hide this error, please tell.

Comment (igordmn) [reply scripts/mergeAOSP.sh:46]: collection/**/nativeTest should be picked from AOSP

Comment (MatkovIvan) [reply scripts/mergeAOSP.sh:46]: > If you know how to hide this error, please tell.

` > /dev/null` but it doesn't sound right to mute all output here, so let's keep it as-is

**Pull request merged**